### PR TITLE
Replace Apache Commons Math3 with Maven dependency

### PR DESCRIPTION
**Pull request by Noreaster76**
Replace the copied-in code for the `org.apache.commons.commons-math3` library with that of a Maven dependency (of the same library).

Benefits:
- Less code in the codebase that needs to be maintained.
- Less code for a newcomer to get acquainted with.
- Much easier to incorporate any updates to the `org.apache.commons.commons-math3` library -- it's a simple matter of updating the version in the Maven POM (assuming no breaking changes, of course).

Obviously, the `org.apache.commons.commons-math3` code is always available on Apache's repo to look at, if desired.

### CODE REVIEW

File: src/main/java/new_metrics/CustomBetaDistribution.java [hunk 1]
```diff
@@ -25,20 +10,18 @@
...
-import new_metrics.*;
-
 /**
- * Implements the Beta distribution.
+ * Implements the Beta distribution. Customized from org.apache.commons.commons-math3 BetaDistribution.
  *
  * @see <a href="http://en.wikipedia.org/wiki/Beta_distribution">Beta distribution</a>
  * @since 2.0 (changed to concrete class in 3.0)
  */
-public class BetaDistribution extends AbstractRealDistribution {
```
Comment (Noreaster76) [inline src/main/java/new_metrics/CustomBetaDistribution.java:36]: I think we could eventually compose a `CustomBetaDistribution` object with a `BetaDistribution` object so as to delegate as much logic from the former to the latter and not have to maintain a separate copy. But that can be done in a future PR.

File: src/main/java/new_metrics/CustomGammaDistribution.java [hunk 2]
```diff
@@ -24,12 +9,12 @@
 import org.apache.commons.math3.util.FastMath;
 
 /**
- * Implementation of the Gamma distribution.
+ * Implementation of the Gamma distribution. Customized from org.apache.commons.commons-math3 GammaDistribution.
  *
  * @see <a href="http://en.wikipedia.org/wiki/Gamma_distribution">Gamma distribution (Wikipedia)</a>
  * @see <a href="http://mathworld.wolfram.com/GammaDistribution.html">Gamma distribution (MathWorld)</a>
  */
-public class GammaDistribution extends AbstractRealDistribution {
```
Comment (Noreaster76) [inline src/main/java/new_metrics/CustomGammaDistribution.java:32]: I think we could eventually compose a `CustomGammaDistribution` object with a `GammaDistribution` object so as to delegate as much logic from the former to the latter and not have to maintain a separate copy. But that can be done in a future PR.

**Comment by happyjack27** [comment]
I think I might have copied in the code so that I could do some custom
modifications, to do some bayesian probability distributions, for the
"advanced stats".  I'll have to check.

**Comment by happyjack27** [comment]
Ah, I see you extended the classes I modified and put my new / modified code in the extended class.  I was thinking the same thing.  Looks good.

**Pull request merged**